src/utils/query/createExtensionStoragePersister.ts

```diff
@@ -0,0 +1,36 @@
+import type { PersistedClient, Persister } from "@tanstack/query-persist-client-core";
+import { ExtensionStorage } from "~utils/storage";
+
+/**
+ * Creates a persister that uses ExtensionStorage for React Query cache persistence
+ * @param options Configuration options
+ * @param options.cacheKey Unique key to use for storing the cache in ExtensionStorage
+ * @returns A persister object that can be used with persistQueryClient
+ */
+export function createExtensionStoragePersister({ cacheKey }: { cacheKey: string }): Persister {
+  return {
+    persistClient: async (persistClient: PersistedClient) => {
+      try {
+        await ExtensionStorage.set(cacheKey, JSON.stringify(persistClient));
+      } catch (error) {
+        console.error(`Error persisting query client (${cacheKey}):`, error);
+      }
+    },
+    restoreClient: async () => {
+      try {
+        const cached = await ExtensionStorage.get<string>(cacheKey);
+        return cached ? (JSON.parse(cached) as PersistedClient) : null;
```

Comment:
You don't need to `JSON.parse` as its handled internally.

```suggestion
        const cached = await ExtensionStorage.get<PersistedClient>(cacheKey);
        return cached ?? null;
```

Reply:
Addressed in ef9a16af21734b74fd0ddacb5c7ea15a195e33a4